test/integration/ethereum/viewer/setTokenRateViewerIntegration.spec.ts

```diff
@@ -0,0 +1,88 @@
+import "module-alias/register";
+import { ethers, network } from "hardhat";
+import { Account } from "@utils/types";
+import DeployHelper from "@utils/deploys";
+import { getAccounts, getWaffleExpect, ether } from "@utils/index";
+
+import { IERC20, ISetToken, SetTokenRateViewer } from "../../../../typechain";
+import { PRODUCTION_ADDRESSES, STAGING_ADDRESSES } from "../addresses";
+
+const expect = getWaffleExpect();
+
+if (process.env.INTEGRATIONTEST) {
```

Comment:
Is there a case here where we test the logic when the position is negative ? if not I suggest adding it.

Reply:
Yep there's a case for negative positions